design/secrets_provider_milestone1_design.md

```diff
@@ -0,0 +1,474 @@
+
+### Requirements
+
+The solution should support the following capabilities:
+
+Milestone 1 *(current)*
+
+- Secrets Provider runs and deployed without affecting the app deployment or app life cycle
+- Secrets Provider can serve many apps
+
+Milestone 2
+
+- Secrets Provider support secret rotation
+
+### Future requirements
+
+The solution should consider the following future requirements that may arise:
+
+* Support other **targets** for the secrets besides K8s Secrets, such as files
+* Support other **sources** for secrets paths besides conjur-map in K8s Secrets
+* Support **sources** varying while Secrets Provider runs
+
+## Solution
+The solution is to allow different deployments and behave differently based on the use-case chosen.
+
+The following decision tree depicts customer uses-cases for Secrets Provider deployments:
+
+![Secrets Provider flavors desicion flow chart](https://user-images.githubusercontent.com/31179631/85747023-975bf500-b70f-11ea-8e26-1134068fe655.png)
+
+This decision tree shows 6 different variations of Secrets Provider:
+
+| Secrets Provider variation                            | Business value                                               | Implementation status |
+| ----------------------------------------------------- | ------------------------------------------------------------ | --------------------- |
+| Run as **Init Container** and sync to **K8s Secrets** | Supply secrets on application load, based on K8s secrets     | Implemented           |
+| Run as **Job** and sync to **K8s Secrets**            | Supply secrets once for many applications, based on K8s secrets configurations | Milestone 1           |
+| Run as **Deployment** and sync to **K8s Secrets**     | Supply secrets for many applications and support rotation, based on K8s secrets | Future (Milestone 2)  |
+| Run as **Sidecar** and sync to **K8s Secrets**        | Support secrets rotation, based on K8s secrets               | Future (Milestone 2)  |
+| Run as **Init Container** and sync to **files**       | Supply secrets on application load, pushed as files to shared volume | Future (Milestone 3)  |
+| Run as **Sidecar** and sync to **files**              | Supply secrets for one application and support rotation, pushed as files to shared volume | Future (Milestone 3)  |
+
+These variations will be supported using the same Secrets Provider image, but the behavior will vary dynamically depending on the chosen deployment.
+
+*In this document we will focus on **Milestone 1**.*
+
+
+
+### Milestone 1: Serve K8s secrets to multiple applications once (no rotation)
+
+#### Design
+
+Run Secrets Provider as a **Job**.
+Once the Job's pod is spun up, it will authenticate to Conjur/DAP via authn-k8s.
+It will then fetch all the K8s secrets denoted with a specific label and update them with the Conjur secrets they require and upcome completion, terminate.
```

Comment:
you say here that it `will then fetch all the K8s secrets denoted with a specific label` but later in the doc, under "Customer Experience" you say that the label is optional. what happens if the label is not there?

Reply:
also - can you link to the API you will use?

Reply:
I will add something like the following to the doc
Check if K8S_SECRETS exists in manifest, 
  Yes? Take the K8S Secrets values from there and fetch secrets
  No? Check K8S_SECRETS_LABEL exists
    Yes? Fetch secrets
     No? Fail

Reply:
API is referenced further down. But here you go: https://docs.openshift.com/container-platform/4.4/rest_api/index.html#list-14

Reply:
thanks, couldn't find it in the doc.